calyx-opt/src/passes/static_inliner.rs

```diff
@@ -0,0 +1,388 @@
+use crate::analysis::ControlId;
+use crate::traversal::{Action, Named, VisResult, Visitor};
+use calyx_ir as ir;
+use calyx_ir::structure;
+use calyx_ir::LibrarySignatures;
+use ir::build_assignments;
+use std::collections::HashMap;
+use std::rc::Rc;
+
+#[derive(Default)]
+pub struct StaticInliner {
+    /// maps ids of static control stmts to groups
+    map: HashMap<u64, ir::RRC<ir::StaticGroup>>,
+}
+
+impl Named for StaticInliner {
+    fn name() -> &'static str {
+        "static-inline"
```

Comment:
This is why I recommend using the `::name` function. The error message above says `static-inliner` but the pass is called `static-inline`.